pxr/usd/bin/usdtree/usdtree.py

```diff
@@ -0,0 +1,228 @@
+    else:
+        return sorted(prim.ListInfoKeys())
+
+def _GetName(prim, iterType):
+    if iterType == 'UsdPrim':
+        return prim.GetName()
+    else:
+        return prim.name
+
+def _GetTypeName(prim, iterType):
+    if iterType == 'UsdPrim':
+        return prim.GetTypeName()
+    else:
+        return prim.typeName
+
+
+def PrintPrim(args, prim, iterType, prefix, isLast):
+    if not isLast:
+        lastStep = ' |--'
+        if _GetChildren(prim, iterType):
+            attrStep = ' |   |'
+        else:
+            attrStep = ' |    '
+    else:
+        lastStep = ' `--'
+        if _GetChildren(prim, iterType):
+            attrStep = '     |'
+        else:
+            attrStep = '      '
+    if args.types:
+        typeName = _GetTypeName(prim, iterType)
+        if typeName:
+            label = '{}[{}]'.format( _GetName(prim, iterType), typeName)
+        else:
+            label =  _GetName(prim, iterType)
+    else:
+        label =  _GetName(prim, iterType)
+    _Msg('{}{}{}'.format(prefix, lastStep, label))
+
+    attrs = []
+    if args.metadata:
+        attrs.extend('({})'.format(md) for md in _GetMetadata(prim, iterType))
+    
+    if args.attributes:
+        attrs.extend('.{}'.format(prop.GetName()) for prop in _GetProperties(prim, iterType))
+    
+    numAttrs = len(attrs)
+    for i, attr in enumerate(attrs):
+        if i < numAttrs - 1:
+            _Msg('{}{} :--{}'.format(prefix, attrStep, attr))
+        else:
+            _Msg('{}{} `--{}'.format(prefix, attrStep, attr))
+
+
+def PrintChildren(args, prim, iterType, prefix):
+    children = _GetChildren(prim, iterType)
+    numChildren = len(children)
+    for i, child in enumerate(children):
+        if i < numChildren - 1:
+            PrintPrim(args, child, iterType, prefix, isLast=False)
+            PrintChildren(args, child, iterType, prefix + ' |  ')
+        else:
+            PrintPrim(args, child, iterType, prefix, isLast=True)
+            PrintChildren(args, child, iterType, prefix + '    ')
+
+
+def PrintStage(args, stage):
+    _Msg('USD')
+    PrintChildren(args, stage.GetPseudoRoot(), 'UsdPrim', '')
+
+
+def PrintLayer(args, layer):
+    _Msg('USD')
+    PrintChildren(args, layer.pseudoRoot, 'SdfPrimSpec', '')
+
+
+def PrintTree(args, path):
+    if args.flatten:
+        from pxr import Usd
+        popMask = (None if args.populationMask is None else Usd.StagePopulationMask())
+        if popMask:
+            for mask in args.populationMask:
+                popMask.Add(mask)
+        if popMask:
+            if args.unloaded:
+                stage = Usd.Stage.OpenMasked(path, popMask, Usd.Stage.LoadNone)
+            else:
+                stage = Usd.Stage.OpenMasked(path, popMask)
+        else:
+            if args.unloaded:
+                stage = Usd.Stage.Open(path, Usd.Stage.LoadNone)
+            else:
+                stage = Usd.Stage.Open(path)
+        if args.flattenLayerStack:
```

Comment:
A neat idea to try to combine population masking with layerStack flattening! Unfortunately, our doc is a bit misleading for UsdUtilsFlattenLayerStack: it doesn't use the input stage in any meaningful way - it only grabs the root layerStack out of it, which is unaffected by population masking.
Further, in other tools, --flatten and --flattenLayerStack are independent command-line args, so there's no reason to make them dependent here.

And when you open a Stage in the flattenLayerStack case, you can *always* open it unloaded, because load-state also does not affect the root layerStack.